### PR TITLE
With Apollo GraphQL Example

### PR DESCRIPTION
**Pull request by ebrahim-2**
Add an example of Razzle with Apollo GraphQL

### CODE REVIEW

File: examples/with-react-apollo/src/lib/apollo.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    connectToDevTools: process.browser,
+    ssrMode: !process.browser,
+    ssrForceFetchDelay: 100,
+    link: createHttpLink({
+      uri: "https://metaphysics-production.artsy.net/",
+      fetch
+    }),
+    defaultOptions: {
+      query: {
+        fetchPolicy: "cache-and-network"
```
Comment (yoalix) [inline examples/with-react-apollo/src/lib/apollo.js]: query's fetchPolicy does not have "cache-and-network". watchQuery on the other hand does contain "cache-and-network".

File: examples/with-react-apollo/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "graphql": "^14.3.1",
+    "graphql-tag": "^2.10.1",
+    "isomorphic-unfetch": "^3.0.0",
+    "razzle": "^3.0.0",
+    "react": "^16.8.6",
+    "react-apollo": "^2.5.6",
+    "react-dom": "^16.8.6",
+    "react-router-dom": "^5.0.0"
+  }
+}
```
Comment (adaam2) [inline examples/with-react-apollo/package.json]: New line at EOF please

**Review by adaam2 — APPROVED** [review]

**Comment by kkalavantavanich** [comment]
Is this the same as #950?

File: examples/with-react-apollo/README.MD [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+Install it and run:
+
+```bash
+yarn install
+yarn start
+```
+
+## Idea behind the example
+This is a basic, bare-bones example of how to use razzle with apollo.
```
Comment (jaredpalmer) [inline examples/with-react-apollo/README.MD]: This could use some more information. Also should have a link to Apollo repo and docs

**Review by jaredpalmer — CHANGES_REQUESTED** [review]
Thank you for this! 

Overall, this looks a bit more opinionated than I think we want for a "bare bones" example. A few things to change/discuss.

- Should we consider using `apollo-boost` for the "bare-bones" example and maybe add another example called `with-apollo-client` or like `with-apollo-advanced`? cc @peggyrayzis 
- Add comments in areas where people should change up variables
- Elaborate what is going on in the README. One sentence about what Apollo is and then a paragraph about this gestault of this example--calling out what folks will want to change as well.
-

File: examples/with-react-apollo/README.MD [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Razzle With Apollo Example
+
+## How to use
+Download the example [or clone the whole project](https://github.com/jaredpalmer/razzle.git):
+
+```bash
+curl https://codeload.github.com/jaredpalmer/razzle/tar.gz/master | tar -xz --strip=2 razzle-master/examples/with-react-apollo
```
Comment (matiasherranz) [inline examples/with-react-apollo/README.MD]:
```
curl https://codeload.github.com/jaredpalmer/razzle/tar.gz/master | tar -xz --strip=2 razzle-master/examples/with-react-apollo
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  626k    0  626k    0     0   334k      0 --:--:--  0:00:01 --:--:--  334k
tar: razzle-master/examples/with-react-apollo: Not found in archive
tar: Error exit delayed from previous errors.
```

**Comment by fivethreeo** [comment]
Could you do this against the canary release and use the razzle graphql plugin?

**Comment by matiasherranz** [comment]
> Could you do this against the canary release and use the razzle graphql plugin?

What do you mean with " the canary release"? If provided with some context, I'll be happy to help/provide more info/test again.

Thanks for the follow up!

**Comment by fivethreeo** [comment]
There is a development release now. A canary that is a development release for 4.x with webpack 5 support. I prioritize new features and support for 4.x. So if this is done against the canary branch in git it will be forwards compatible. To get a basic example up and running: 

```bash
npx create-razzle-app@canary my-app
cd my-app
yarn start 
```

**Comment by fivethreeo** [comment]
To try it now:

```bash
npx create-razzle-app@canary testing --ignore-existing
cd testing
yarn add webpack@5.0.0-rc.3 --dev
yarn start
yarn build
yarn start:prod
yarn test
```

[See the docs](https://razzle-git-canary.jared.vercel.app/getting-started)

**Pull request closed without merging**